test/test_pebble.py

```diff
@@ -418,6 +418,9 @@ def test_yaml(self):
 services:
   bar:
     command: echo bar
+    environment:
+    - ENV1: value1
+    - ENV2: value2
```

Comment:
I see this change to a raw YAML response, but I don't see any corresponding test that shows that we are parsing this response correctly. eg:
`self.assertEqual(s.environment, [{'ENV1': 'value1'}, {'ENV2': 'value2'}])`

I guess we are testing that we can round trip the content to yaml and back again, but I would expect to test python equivalent (in python we have a list of dicts)

Reply:
Good call -- added. Though note that part of this change was to change it to the (IMO more Pythonic) list of 2-tuples, so `[('ENV1', 'value1'), ('ENV2', 'value2')]`.